neqo-qpack/src/encoder.rs

```diff
@@ -495,15 +495,14 @@ impl QPackEncoder {
             .write_byte(QPACK_UNI_STREAM_TYPE_ENCODER as u8);
     }
 
-    pub fn add_recv_stream(&mut self, stream_id: u64) {
-        if self.remote_stream_id.is_some() {
-            panic!("Adding multiple remote streams");
+    pub fn add_recv_stream(&mut self, stream_id: u64) -> Res<()> {
```

Comment:
If `add_recv_stream` now returns Res for Error::WrongStreamCount, then callers (e.g. Http3Connection::decode_new_stream) do not also need to convert the Err to the same WrongStreamCount error.

Reply:
This is because they are not the same type neqo_qpack::Error::WrongStreamCount anf neqo_http3::Error::WrongStreamCount. I can fix this in a different way as well, I will do it in a follow up.